packages/types/api.d.ts

```diff
@@ -2,6 +2,11 @@ export as namespace chevrotain
 
 export declare const VERSION: string
 
+export type ParserMethod<ARGS extends unknown[], R> = ((...args: ARGS) => R) & {
```

Comment:
did you combine the `occurrenceIdx` and the  regular rule parameters into the variable args ARGS?

Won't this approach lose type information for a consumer?

Reply:
Ignore the previous comment, I later realized this refactor removes the need for the idx 👍